### DISCUSSION TITLE
How to scan cluster and code for changes due to flux upgrade

### DISCUSSION THREAD

**Discussion by kondil** [fluxcd/flux2#5054] [Q&A]
Hey guys,

sorry for my newbie question ... I just don't know what to search and how to search for it ...

**Scenario**

I have a cluster running flux `v2.3.0`.

I want to upgrade flux to `v2.4.0`

**Questions**

1. Is there a `flux` way to scan my cluster for what is changing between the 2 versions?
2. Is there a `flux` way to scan my code/repos for what is changing between the 2 versions?

**Clarifications**

Right now I will read the CHANGELOG, create a small script that will grep the code/repos or fetch all objects' configuration from my cluster and grep for the required changes.

What I am looking for is whether `flux` provides a more elegant, out of the box functionality to achieve this.

If this is documented already, the link to it would be very much appreciated.

**Comment by stefanprodan** [comment]
We have a GitHub Action that will open a PR when there is a new Flux version available: https://fluxcd.io/flux/flux-gh-action/#automate-flux-updates

**Comment by kondil** [comment]
Hey @stefanprodan,

thank you for this. I have been checking this already for automating our flux upgrade procedure.

However what I am asking above is how can I scan my cluster's components manifests for required changes  according to the upcoming flux upgrade.

For example, in [flux v2.4.0](https://github.com/fluxcd/flux2/releases/tag/v2.4.0) the [Bucket](https://fluxcd.io/flux/components/source/buckets/) kind was promoted from `v1beta2` to `v1` (GA).

So what I am looking for is something that I can run against my yaml manifests or my cluster and for any bucket object not in `v1` to throw me some notification.

Not sure if such thing exists already or if I am asking for something not available.
